src/main/java/forestry/apiculture/genetics/effects/NonStackingBeeEffect.java

```diff
@@ -95,32 +97,28 @@ private void performGlobalEffect(TickEvent.LevelTickEvent event) {
 		}
 	}
 
-	private void applyEffectToNearbyTiles(HashSet<BlockPos> affectedHives, Level level, BlockPos pos, Vec3i territory) {
-		int x = pos.getX();
-		int z = pos.getZ();
-		int xWidth = territory.getX();
-		int zWidth = territory.getZ();
-		int xChunksPositive = (x % 16 + xWidth) / 16;
-		int zChunksPositive = (z % 16 + zWidth) / 16;
-		int xChunksNegative = (x % 16 - xWidth) / 16;
-		int zChunksNegative = (z % 16 - zWidth) / 16;
-		int chunkX = SectionPos.blockToSectionCoord(pos.getX());
-		int chunkZ = SectionPos.blockToSectionCoord(pos.getZ());
-
-		for (int i = chunkX - xChunksPositive; i <= chunkX + xChunksNegative; i++) {
-			for (int j = chunkZ - zChunksPositive; j <= chunkZ + zChunksNegative; j++) {
-				level.getChunk(i, j).blockEntities.forEach((targetPos, blockEntity) -> {
-					if (blockEntity instanceof IBeeHousing housing) {
-						int distX = Math.abs(pos.getX() - targetPos.getX());
-						int distY = Math.abs(pos.getY() - targetPos.getY());
-						int distZ = Math.abs(pos.getZ() - targetPos.getZ());
-
-						if (distX > territory.getX() || distY > territory.getY() || distZ > territory.getZ() || pos.equals(targetPos)) {
+	private void affectNearbyTiles(HashSet<BlockPos> affectedHives, Level level, BlockPos pos, Vec3i territory) {
+		BlockPos topLeft = pos.offset(VecUtil.center(territory));
+		BlockPos bottomRight = topLeft.offset(territory);
+		for (int x = SectionPos.blockToSectionCoord(topLeft.getX()); x <= SectionPos.blockToSectionCoord(bottomRight.getX()); x++) {
+			for (int z = SectionPos.blockToSectionCoord(topLeft.getZ()); z <= SectionPos.blockToSectionCoord(bottomRight.getZ()); z++) {
+				level.getChunk(x, z).getBlockEntities().forEach((targetPos, tileEntity) -> {
```

Comment:
Should we check if the chunk is loaded before accessing it? I didn't do this in my original version but it might be a good idea.